File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Expressions/ArithmeticExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model.ADT;
namespace LabAssig7.Model.Expressions
{
    public class ArithmeticExpression:Expression
    {
        private char op;
        private Expression left, right;

        public ArithmeticExpression(char oper,Expression l,Expression r)
        {
            this.op = oper;
            this.left = l;
            this.right = r;
        }

        public int Eval(ISymbolTable<string,int> dict)
        {
            int resl = this.left.Eval(dict);
            int resr = this.right.Eval(dict);

            if (this.op == '+')
                return resl + resr;
            else if (this.op == '-')
                return resl - resr;
            else if (this.op == '*')
                return resl * resr;
            else if(this.op == '%')
            {
                if (resr == 0)
                    throw new DivideByZeroException("Division by 0");
                else
                    return resl % resr;
            }
            else
            {
                if (resr == 0)
                    throw new DivideByZeroException("Division by 0");
                else
                    return resl / resr;
            }
        }

        public override string ToString()
        {
            return ""+this.left + this.op + this.right;
        }
    }
}

File: MAP(Advanced Programming Methods)/lab2v2/src/Model/ArithmeticExpression.java
package Model;
import Exceptions.*;
public class ArithmeticExpression implements Expression {

    private char operator;
    private Expression left, right;

    public ArithmeticExpression(char operator, Expression left, Expression right)
    {
        this.operator=operator;
        this.left=left;
        this.right=right;
    }

    public int eval(IDictionary<String,Integer> dict)
    {
        int left=this.left.eval(dict);
        int right=this.right.eval(dict);
        if(this.operator == '+')
            return left+right;
        else if(this.operator == '-')
            return left-right;
        else if(this.operator == '*')
            return left*right;
        else if(this.operator == '%')
            return left%right;
        else
            if(right == 0)
                throw new DivisionByZeroException("We have a division by zero");
            else
                return left/right;

    }

    @Override
    public String toString()
    {
        return ""+left+this.operator+right;
    }
}
